From 6e9e84ddb5a052f6a8d4ba3e1cb843eb605be099 Mon Sep 17 00:00:00 2001
From: Deepansh Rastogi <deepansh.rastogi@intel.com>
Date: Mon, 6 Jan 2020 20:22:52 +0530
Subject: [PATCH] UGW_SW-44604: Clearing flow id bit in GSWIP rx to select
 correct egress Q based on flowid bits in VDSL bonding tx.

---
 drivers/net/datapath/dpm/gswip30/datapath_rx.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/datapath/dpm/gswip30/datapath_rx.c b/drivers/net/datapath/dpm/gswip30/datapath_rx.c
index 27107b0c2987..71e9a681d3d5 100644
--- a/drivers/net/datapath/dpm/gswip30/datapath_rx.c
+++ b/drivers/net/datapath/dpm/gswip30/datapath_rx.c
@@ -245,6 +245,7 @@ int32_t dp_rx_30(struct sk_buff *skb, u32 flags)
 					desc_0, desc_1,
 					desc_2, desc_3);
 		}
+		desc_0->field.flow_id = 0;
 #if IS_ENABLED(CONFIG_INTEL_DATAPATH_MPE_FASTHOOK_TEST)
 		if (unlikely(ltq_mpe_fasthook_rx_fn))
 			ltq_mpe_fasthook_rx_fn(skb, 1, NULL);	/*with pmac */
